docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md

```diff
@@ -0,0 +1,109 @@
+## Sentence Processing and Feature Extraction Techniques for Ice Cream Selling
+
+This document discusses the possible sentence processing and feature extraction techniques 
+applied with NLP module of ravestate library for ice cream selling.
+
+#### NLP - Extracted Features
+
+| Feature                           | Ravestate Properties/Signals  | Description                                                 | Example Sentence: 'Revolutions need six fancy chickens!'|
+| -------------                     | --------------------          |-------------------------------                              | ------------------------------|
+| Tokenization                      | `nlp.prop_tokens`             | Segmenting text into words, punctuation marks etc.          | 'Revolutions', 'need', 'six', 'fancy', 'chickens', '!'|
+| Part-of-Speech (POS) Tagging      | `nlp.prop_postags`            | Assigning word types to tokens                              | 'NOUN', 'VERB', 'NUM', 'ADJ', 'NOUN', 'PUNCT' |
+| Detailed POS Tag                  | `nlp.prop_tags`               | Fine-grained part-of-speech                                 | 'Revolutions' has the tag: 'NNS', which stand for: noun, plural <br> 'need' has the tag: 'VBP', which stands for: verb, non-3rd person singular present <br> [List](https://spacy.io/api/annotation#pos-tagging) of POS tags|
+| Lemmatization                     | `nlp.prop_lemmas`             | Assigning the base forms of words                           | 'Revolutions' has the lemma: 'revolution' <br>  'was' would have the lemma: 'be'|
+| Named Entity Recognition (NER)    | `nlp.prop_ner`                | Labelling "real-world" objects ("NE"=Named Entity)          | 'six' has the NE: 'CARDINAL', which are numerals that do not fall under another type <br> [List](https://spacy.io/api/annotation#named-entities) of NEs|
+| Triple Extraction                 | `nlp.prop_triples`            | A triple consists of subject, predicate, object of sentence | Triple: subject: 'Revolutions', predicate: 'need', object: 'chickens' |
+| About Roboy                       | `nlp.prop_roboy`              | Detecting whether sentence is about Roboy                   | 'you', 'roboy', 'robot', 'roboboy', ... |
+| Yes-No                            | `nlp.prop_yesno`              | Detecting answers to yes-no questions                       | Checking for 'yes', 'no', 'i don't know', 'probably', 'probably not' and synonyms of these                           |
+| Sentence Type: Question           | `nlp.sig_is_question`         | Emitted if the input sentence is a question                 |                                                |
+| Play Game                         | `nlp.sig_intent_play`         | Emitted when the interlocutor wants to play a game          | input: "I want to play", "I like games" or something similar    |
+
+### Using the Features
+
+#### Detailed POS Tagging
```

Comment:
This is not entirely correct - Triples are already build on top of this. For example, if you want to match an ice cream flavor, you can use `ctx[nlp.prop_triples][0].match_either_lemma(obj={"vanilla", "strawberry", "lemon", ...})`.

If you want to detect questions, this is also built into triple - use `ctx[nlp.prop_triples][0].is_question()`, and the get the question word either from `ctx[nlp.prop_triples][0].get_subject()` or `ctx[nlp.prop_triples][0].get_object()`.

Reply:
I also thought about this solution but Triples is not working as i expected, it only captures the last word of the corresponding subject. For example if the string is 'i want vanilla ice cream', it extracts '[<Triple object i:want:cream>]'. On the other hand we can understand whether there is a request or not. Therefore i was planning to get the customer's intention with Triples and flavors with POS Tagging.